autoload/vim_ai_config.vim

```diff
@@ -74,6 +78,10 @@ let g:vim_ai_chat_default = {
 \    "code_syntax_enabled": 1,
 \    "paste_mode": 1,
 \  },
+\  "provider": {
```

Comment:
I was thinking about introducing a global provider registry.
The extension plugin would need to call it to register its provider(s), e.g.:

```vim
vim_ai#RegisterAIProvider('bedrock', other_provider_options)
```

And the user would simply reference it by the name:

```vim
let g:vim_ai_edit_default = {
\  "provider": "bedrock",
\  "options" : bedrock_options,
\}
```

Reply:
What would `other_provider_options` be providing to the registration function?

Reply:
Things needed for registration. Like class name, path to module